cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.md

```diff
@@ -157,15 +159,24 @@ public class ForwardServlet extends HttpServlet
 Safe use of redirects and forwards can be done in a number of ways:
 
 - Simply avoid using redirects and forwards.
-- If used, do not allow the url as user input for the destination. This can usually be done. In this case, you should have a method to validate URL.
+- If used, do not allow the URL as user input for the destination. This can usually be done. In this case, you should have a method to validate URL.
 - If user input can’t be avoided, ensure that the supplied **value** is valid, appropriate for the application, and is **authorized** for the user.
 - It is recommended that any such destination input be mapped to a value, rather than the actual URL or portion of the URL, and that server side code translate this value to the target URL.
-- Sanitize input by creating a list of trusted URL's (lists of hosts or a regex).
+- Sanitize input by creating a list of trusted URLs (lists of hosts or a regex).
 - Force all redirects to first go through a page notifying users that they are going off of your site, and have them click a link to confirm.
 
+## Validating URLs
+When attempting to validate and sanitise user-input to determine whether the URL is safe, wherever possible you should use a built in library or function to parse the URLs, such as `parse_url()` in PHP, rather than rolling your own parser using regex. Additionally, make sure that you take the following into account:
+
+- Input starting with a `/` to redirect to local pages is **not safe**. `//example.org` is a valid URL.
+- Input starting with the desired domain name is **not safe**. `https://example.org.attacker.com` is valid.
+- JavaScript URIs such as `javascript:alert(1)` should be blocked
```

Comment:
I will advise to only allow the HTTP/HTTPS protocol to be sure that no other protocol are used like Javascript/Ftp/...

Reply:
Added this as a point.